tests/test_unit_command.cc

```diff
@@ -1703,8 +1703,327 @@ namespace sc2 {
         }
     };
 
+    //
+    // TestNoTarget_Tag
+    //
+
+    class TestNoTarget_Tag : public TestUnitCommandNoTarget {
```

Comment:
Yes, I think we can extend the existing tests as you suggest.  I think I latched on to the first solution that presented itself to me, but the tests logic becomes clearer to me the more I play with it.  It was good practice though ;)

Reply:
With current approach that UnitCommand(Unit) will eventually call UnitCommand(Tag), does the test need to explicitly call UnitCommand(Tag) or is it sufficient that it gets called in the logic?

In `test_unit_command_common.cc` the function `IssueUnitCommand(ActionInterface*)` is setup to take Unit* or Units overload.
``` 
293        if (test_units_.size() > 1) {
294            act->UnitCommand(test_units_, test_ability_, target_unit_);
295        }
296        else {
297            act->UnitCommand(test_unit_, test_ability_, target_unit_);
298        }

// *.h
 24       const Unit* test_unit_ = nullptr;
 25       Units test_units_;
```

I think that was part of the reason I had originally went with class inheritance.  I could not think of an elegant way to change the `test_units_` data member type or substitute in another type, without recreating the function.

It makes implementation simpler and less code duplication if the Tag overload doesn't have to be explicitly called.  In that case, it would simply require making sure `TestUnitCommandTargetingUnit` , `TestUnitCommandTargetingPoint` , and `TestUnitCommandNoTarget` get used at least once as Units and Unit*.

Reply:
Speaking about best practices: we should test both versions of `UnitCommand` independently because methods could be much more complex or different in the future (consider e.g. addition of another entity which will issue request action with unit or with tag). Thus, I suggest to assume in the tests that we don't know which method calls which inside.

Speaking about `test_units_` I would try the following:
a. Take list of units.
b. Give the whole list some command (e.g. move or hold).
c. Convert the list to list of tags.
d. Give the new list another command.

You don't have to store additional list in the class, all we need is to check that the command was issued and executed.

Reply:
Ok, lets drop the new tests and extend the existing Units command test if needed (e.g. by adding new commands or trying to pass nullptrs). Should be ok as we use the new functionality below.